test/uk/gov/hmrc/helptosaveproxy/util/AuthSupport.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosaveproxy.util
+
+import uk.gov.hmrc.auth.core.AuthProvider.{GovernmentGateway, PrivilegedApplication}
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthProviders, AuthorisedFunctions}
+import uk.gov.hmrc.auth.core.authorise.Predicate
+import uk.gov.hmrc.auth.core.retrieve
+import uk.gov.hmrc.auth.core.retrieve._
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.helptosaveproxy.TestSupport
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
+import uk.gov.hmrc.helptosaveproxy.auth.Auth
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait AuthSupport extends TestSupport {
+
+  val nino = "AE123456C"
+
+  val ggCredentials = GGCredId("123-gg")
+
+  val privilegedCredentials = PAClientId("123-pa")
+
+  val authProviders: AuthProviders = AuthProviders(GovernmentGateway, PrivilegedApplication)
+
+  val mockAuthConnector: AuthConnector = mock[AuthConnector]
+
+  class AuthConn extends Auth(mockAuthConnector) with AuthorisedFunctions with BaseController with Logging with WithMdcExecutionContext
```

Comment:
why do have to create this class ? cant we just mock ? take a look at the frontend `AuthSupport` class on how we mock

Reply:
I have renamed this to TestAuth and placed it in the AuthSpec as thats the only place its needed. It seems to be difficult to mock the Auth class as it has a self type.